module/VuFind/src/VuFind/Tags.php

```diff
@@ -74,4 +73,38 @@ public function parse($tags)
         }
         return array_unique($result);
     }
+
+    /**
+     * Add tags to the record.
+     *
+     * @param RecordDriver        $driver Driver representing record being tagged
+     * @param UserEntityInterface $user   The user adding the tag(s)
+     * @param string[]            $tags   The user-provided tag(s)
+     *
+     * @return void
+     */
+    public function addTagsToRecord(RecordDriver $driver, UserEntityInterface $user, array $tags): void
+    {
+        $resource = $this->resourcePopulator->getOrCreateResourceForDriver($driver);
+        foreach ($tags as $tag) {
+            $resource->addTag($tag, $user);
```

Comment:
This (and the deleteTag) method below will need to be refactored into \VuFind\Db\Service\TagService... but that's a project for a separate PR!